bee-storage/bee-storage-rocksdb/src/access/stream.rs

```diff
@@ -0,0 +1,350 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    column_families::*,
+    storage::{Storage, StorageBackend},
+    system::System,
+};
+
+use bee_common::packable::Packable;
+use bee_ledger::{
+    snapshot::info::SnapshotInfo,
+    types::{Balance, ConsumedOutput, CreatedOutput, LedgerIndex, OutputDiff, Receipt, TreasuryOutput, Unspent},
+};
+use bee_message::{
+    address::{Address, Ed25519Address},
+    milestone::{Milestone, MilestoneIndex},
+    output::OutputId,
+    payload::indexation::{PaddedIndex, INDEXATION_PADDED_INDEX_LENGTH},
+    Message, MessageId, MESSAGE_ID_LENGTH,
+};
+use bee_storage::access::AsStream;
+use bee_tangle::{
+    metadata::MessageMetadata, solid_entry_point::SolidEntryPoint, unreferenced_message::UnreferencedMessage,
+};
+
+use futures::{
+    stream::Stream,
+    task::{Context, Poll},
+};
+use pin_project::pin_project;
+use rocksdb::{DBIterator, IteratorMode};
+
+use std::{convert::TryInto, marker::PhantomData, pin::Pin};
+
+#[pin_project(project = StorageStreamProj)]
+pub struct StorageStream<'a, K, V> {
+    #[pin]
+    inner: DBIterator<'a>,
```

Comment:
Would this be better named as `iter` or something similar? Since the name `StorageStream` doesn't suggest anything to do with iteration, I think `inner` causes some confusion here.

Reply:
Thing is that usually streams are the async version of iterators.  And we are asynchronously wrapping a sync operation here (because the trait itself is async).